server/src/test/java/org/elasticsearch/cluster/coordination/MessagesTests.java

```diff
@@ -36,59 +35,57 @@ public void testJoinEqualsHashCodeSerialization() {
             randomNonNegativeLong(),
             randomNonNegativeLong()
         );
-        // Note: the explicit cast of the CopyFunction is needed for some IDE (specifically Eclipse 4.8.0) to infer the right type
-        EqualsHashCodeTestUtils.checkEqualsAndHashCode(
+        // Note: the explicit type parameter is needed for some IDE (specifically Eclipse 4.8.0) to infer the right type
```

Comment:
note that we dropped support for Eclipse, so we should not need to retain these explicit types anymore if it was just for Eclipse.